client/checkout/woopay/direct-checkout/woopay-direct-checkout.js

```diff
@@ -0,0 +1,167 @@
+		set_redirect_session_data_success: 'setWoopaySessionResponse',
+	};
+
+	static init() {
+		this.state.iframePostMessage = new Promise( ( resolve ) => {
+			this.listeners.setIframePostMessage = resolve;
+		} );
+
+		this.injectWooPayConnectIframe();
+	}
+
+	static isWooPayEnabled() {
+		return getConfig( 'isWooPayEnabled' );
+	}
+
+	static injectWooPayConnectIframe() {
+		const checkoutButton = document.querySelector(
+			'.wc-proceed-to-checkout .checkout-button'
+		);
+		if ( checkoutButton ) {
+			checkoutButton.parentElement.style.position = 'relative';
+			const hiddenDiv = document.createElement( 'div' );
+			hiddenDiv.style.visibility = 'hidden';
+			hiddenDiv.style.position = 'absolute';
+			hiddenDiv.style.height = '0';
+
+			checkoutButton.parentElement.appendChild( hiddenDiv );
+
+			ReactDOM.render(
+				<WooPayConnectIframe
+					listeners={ this.listeners }
+					actionCallback={ this.actionCallback }
+				/>,
+				hiddenDiv
+			);
+		}
+	}
+
+	static async setTempThirdPartyCookie() {
+		const setTempThirdPartyCookie = new Promise( ( resolve ) => {
+			this.listeners.setTempThirdPartyCookie = resolve;
+		} );
+
+		const postMessage = await this.state.iframePostMessage;
+		postMessage( { action: 'setTempThirdPartyCookie' } );
+
+		// Since iFrame is refreshed, we need to re-init the iframePostMessage promise.
+		this.state.iframePostMessage = new Promise( ( resolve ) => {
+			this.listeners.setIframePostMessage = resolve;
+		} );
+
+		return await setTempThirdPartyCookie;
+	}
+
+	static async isWooPayThirdPartyCookiesEnabled() {
+		const isCookieSet = await this.setTempThirdPartyCookie();
+		if ( ! isCookieSet ) {
+			return false;
+		}
+
+		const isWooPayThirdPartyCookiesEnabled = new Promise( ( resolve ) => {
+			this.listeners.setIsWooPayThirdPartyCookiesEnabled = resolve;
+		} );
+
+		const postMessage = await this.state.iframePostMessage;
+		postMessage( { action: 'getIsThirdPartyCookiesEnabled' } );
+
+		return await isWooPayThirdPartyCookiesEnabled;
+	}
+
+	static async isUserLoggedIn() {
+		const isUserLoggedIn = new Promise( ( resolve ) => {
+			this.listeners.setIsUserLoggedIn = resolve;
+		} );
+
+		const postMessage = await this.state.iframePostMessage;
+		postMessage( { action: 'getIsUserLoggedIn' } );
+
+		return await isUserLoggedIn;
+	}
+
+	static async sendSessionDataToWooPay() {
+		const woopaySessionData = new Promise( ( resolve ) => {
+			this.listeners.setWoopaySessionResponse = resolve;
+		} );
+
+		const woopaySession = await this.getWooPaySessionFromMerchant();
+		const postMessage = await this.state.iframePostMessage;
+		postMessage( {
+			action: 'setRedirectSessionData',
+			value: woopaySession,
+		} );
+
+		const { redirect_url: redirectUrl } = await woopaySessionData;
+
+		return redirectUrl;
+	}
+
+	static getCheckoutRedirectElements() {
+		// TODO: Add other checkout elements, if needed.
+		const checkoutButton = document.querySelector(
+			'.wc-proceed-to-checkout .checkout-button'
```

Comment:
Could we add a selector here now to capture block checkout?

Reply:
Based on our discussion during the call, we should add that selector now as part of this PR to get blocks compatibility right away.

Reply:
Good call. Added in bcb915b0f165b6f44aa3364512b11eda31e3d8ba.